src/commands/run.cr

```diff
@@ -0,0 +1,41 @@
+require "./command"
+
+module Shards
+  module Commands
+    class Run < Command
+      def run(targets, options, run_options)
+        if spec.targets.empty?
+          raise Error.new("Targets not defined in #{SPEC_FILENAME}")
+        end
+
+        # when more than one target was specified
+        if targets.size > 1
+          raise Error.new("Error please specify only one target. If you meant to pass arguments you may use 'shards run target -- args'")
+        end
+
+        # when no target was specified
+        if targets.empty?
+          if spec.targets.size > 1
+            raise Error.new("Error please specify the target with 'shards run target'")
+          else
+            name = spec.targets.first.name
+          end
+        else
+          name = targets.first
+        end
```

Comment:
I'm too rusty to came with a non-convoluted version of these nested conditions 😬 

Better versions are welcome! 😊

Reply:
The implementation looks fine.

But I'm not sure the resulting behaviour is what we should have. At least the discussion in #157 seemed pretty indecisive on that. It might not be that important *what* we pick, but this alternative is at least not my favourite 🤷

Reply:
> but this alternative is at least not my favourite 

Can you rephrase or explain this? Not sure I follow. The scenarios I made were:

1. When no target was indicated and spec contains only one, assume is that target, runs.
2. When no target was indicated and spec contains more than one, errors out

Looking at the comments in #157 is confusing me:

> Yes. No default, unless there is only 1 target.
https://github.com/crystal-lang/shards/issues/157#issuecomment-539667298

> I'm not sure whether the default behaviour is a good idea. shards build defaults to building all. shards run defaults to running one, but only if there is one defined seems off.
https://github.com/crystal-lang/shards/issues/157#issuecomment-540136983

> Always running the first target if no name specified sounds good. That's consistent behaviour.
https://github.com/crystal-lang/shards/issues/157#issuecomment-541431000

> I lean more and more on always specifying the target to run: consistent, no configuration, no documentation, no repeated boring message (that boils down to: you should always specify the target), no surprises (you know what will happen).
https://github.com/crystal-lang/shards/issues/157#issuecomment-541569088

(skipping the _shards isn't a build tool_ dilema)

I'm ok with making the target mandatory, that will definitely simply the above logic, but I thought of `shards run` similar to `cargo run` and the [default target selection](https://doc.rust-lang.org/cargo/commands/cargo-run.html#target-selection)

Either way, again, we can change it, that is why I sent this PR 😀

Reply:
Yeah, we just need to come to a decision in #157. There are contradicting views on that.
Having a PR ready certainly helps to push this forward ;)